web/search.dart

```diff
@@ -251,21 +296,75 @@ void initializeSearch(
   }
 
   void showSuggestions() {
-    if (presentation.hasChildNodes()) {
+    if (searchResults.hasChildNodes()) {
       listBox.style.display = 'block';
       listBox.setAttribute('aria-expanded', 'true');
     }
   }
 
+  // Function that iterates through the categoriesMap and appends it to the given HTML element
+  // The received Element is the one used for search results page or dropdown list in the search bar
+  void iterateCategoriesMap(Element e) {
+    categoriesMap.forEach((keys, values) {
+      if (categoriesMap[keys] != null) {
+        var value = categoriesMap[keys];
+        e.append(value!);
+      }
+    });
+  }
+
+  // Function that creates the content displayed in the main-content element.
+  void searchResultPage(String input) {
+    var mainContent = document.getElementById('dartdoc-main-content');
+
+    if (mainContent == null) {
+      return;
+    }
+    mainContent.text = '';
+
+    var section = document.createElement('section');
+    section.classes.add('search-summary');
+    mainContent.append(section);
+
+    var title = document.createElement('h2');
+    title.innerHtml = 'Search Results';
+    mainContent.append(title);
+
+    var summary = document.createElement('div');
```

Comment:
Optional, since we don't have a lot of code like this, but internally it is very idiomatic to use [cascades](https://dart.dev/guides/language/language-tour#cascade-notation) in these HTML nodes:

```dart
var summary = document.createElement('div')
    ..classes.add('search-summary')
    ..innerHtml = '$suggestionLength results for "$input"';
```

Might shorten the code or make it look more readable.

Reply:
I liked the format and went ahead and changed it to your suggestion. Thank you